src/components/shared/formItems/__tests__/EthAddress.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react'
+
+import userEvent from '@testing-library/user-event'
+
+import { useForm } from 'antd/lib/form/Form'
+
+import { Form } from 'antd'
+
+import { render } from '../../../../../test/test-utils'
+import EthAddress from '../EthAddress'
+import { readProvider } from '../../../../constants/readProvider'
+
+jest.mock('../../../../constants/readProvider')
+
+function TestableEthAddress() {
+  const [testForm] = useForm<{ address: string }>()
+
+  const [, setAddress] = useState('')
+
+  return (
+    <Form form={testForm}>
+      <Form.Item name="address">
+        <EthAddress
```

Comment:
**question**: to make this test as tightly-scoped as possible, couldn't we instead pass a mocked function to `onAddressChange`, and assert that it was called with the new address? I think we can get away with testing `EthAddress` in isolation, without the need for the wrapping `Form`